components/file_upload/index.js

```diff
@@ -13,12 +13,12 @@ import {canUploadFiles} from 'utils/file_utils';
 
 import FileUpload from './file_upload.jsx';
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
     const config = getConfig(state);
     const maxFileSize = parseInt(config.MaxFileSize, 10);
 
     return {
-        currentChannelId: getCurrentChannelId(state),
+        channelId: ownProps.channelId ?? getCurrentChannelId(state),
```

Comment:
Is it feasible to eliminate this fallback and always pass in `channelId`?

Reply:
Yep, looks like we have ready-access to the channel in create_post, so no need for the getCurrentChannelId..